framework/update/internal/appupdatescenario.cpp

```
@@ -186,14 +169,86 @@ Promise<IInteractive::Result> AppUpdateScenario::showServerErrorMsg()
 
 Promise<Ret> AppUpdateScenario::downloadRelease()
 {
-    RetVal<Val> rv = interactive()->openSync("muse://update/app?mode=download");
-    if (!rv.ret) {
-        return processUpdateError(rv.ret.code());
+    io::path_t packagePath = service()->downloadedReleasePath();
+
+    if (packagePath.empty()) {
+        RetVal<Val> rv = interactive()->openSync("muse://update/app?mode=download");
```

Clang (14.0.6)

[warning] variable name 'rv' is too short, expected at least 3 characters

(readability-identifier-length)